src/components/templates/AvatarGrid/AvatarGrid.tsx

```diff
@@ -0,0 +1,324 @@
+import { AvatarGridRoom } from "types/AvatarGrid";
+import Announcement from "./Announcement";
+import { RoomModal } from "./RoomModal";
+import ChatDrawer from "components/organisms/ChatDrawer";
+
+type Props = {
+  venueName: string;
+};
+
+const DEFAULT_COLUMNS = 40;
+const DEFAULT_ROWS = 25;
+
+const AvatarGrid = ({ venueName }: Props) => {
+  const [selectedUserProfile, setSelectedUserProfile] = useState<
+    WithId<User>
+  >();
+
+  const venueId = venueName;
+  const { user, profile } = useUser();
+  const { venue, partygoers } = useSelector((state) => ({
+    partygoers: state.firestore.ordered.partygoers,
+    venue: state.firestore.data.currentVenue,
+  }));
+  const [isRoomModalOpen, setIsRoomModalOpen] = useState<boolean>(false);
+  const [selectedRoom, setSelectedRoom] = useState<AvatarGridRoom | undefined>(
+    undefined
+  );
+
+  const partygoersBySeat: WithId<User>[][] = [];
+  partygoers.forEach((partygoer) => {
+    if (
+      !partygoer?.data ||
+      partygoer.data[venueId] === undefined ||
+      partygoer.data[venueId].row === undefined ||
+      partygoer.data[venueId].column === undefined
+    )
+      return;
+    const row = partygoer.data[venueId].row || 0;
+    const column = partygoer.data[venueId].column || 0;
+    if (!(row in partygoersBySeat)) {
+      partygoersBySeat[row] = [];
+    }
+    partygoersBySeat[row][column] = partygoer;
+  });
+
+  const takeSeat = useCallback(
+    (row: number | null, column: number | null) => {
+      if (!user || !profile) return;
+      const doc = `users/${user.uid}`;
+      const existingData = profile?.data;
+      const update = {
+        data: {
+          ...existingData,
+          [venueId]: {
+            row,
+            column,
+          },
+        },
+      };
+      const firestore = firebase.firestore();
+      firestore
+        .doc(doc)
+        .update(update)
+        .catch(() => {
+          firestore.doc(doc).set(update);
+        });
+    },
+    [profile, user, venueId]
+  );
+
+  const useKeyPress = function (targetKey: string) {
+    const [keyPressed, setKeyPressed] = useState(false);
+
+    function downHandler({ key }: { key: string }) {
+      if (key === targetKey) {
+        setKeyPressed(true);
+      }
+    }
+
+    const upHandler = ({ key }: { key: string }) => {
+      if (key === targetKey) {
+        setKeyPressed(false);
+      }
+    };
+
+    useEffect(() => {
+      window.addEventListener("keydown", downHandler);
+      window.addEventListener("keyup", upHandler);
+
+      return () => {
+        window.removeEventListener("keydown", downHandler);
+        window.removeEventListener("keyup", upHandler);
+      };
+    });
+
+    return keyPressed;
+  };
+
+  const downPress = useKeyPress("ArrowDown");
+  const upPress = useKeyPress("ArrowUp");
+  const leftPress = useKeyPress("ArrowLeft");
+  const rightPress = useKeyPress("ArrowRight");
+
+  useEffect(() => {
+    const currentPosition = profile?.data?.memrisechats;
```

Comment:
This is going to affect ALL venues with this template. We need to make it based on venueId.